src/otx/v2/adapters/torch/lightning/visual_prompt/dataset.py

```diff
@@ -0,0 +1,171 @@
+"""OTX adapters.torch.lightning.visual_prompt.Dataset API."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Iterable
+
+from omegaconf import DictConfig, OmegaConf
+
+from otx.v2.adapters.torch.lightning.dataset import LightningDataset
+from otx.v2.adapters.torch.lightning.visual_prompt.modules.datasets import OTXVisualPromptingDataset
+from otx.v2.adapters.torch.lightning.visual_prompt.modules.datasets.pipelines import collate_fn
+from otx.v2.api.entities.task_type import TaskType, TrainType
+from otx.v2.api.utils.decorators import add_subset_dataloader
+from otx.v2.api.utils.type_utils import str_to_subset_type
+
+if TYPE_CHECKING:
+    from torch.utils.data import DataLoader as TorchDataLoader
+    from torch.utils.data import Dataset as TorchDataset
+    from torch.utils.data import Sampler
+
+SUBSET_LIST = ["train", "val", "test", "predict"]
+
+
+@add_subset_dataloader(SUBSET_LIST)
+class VisualPromptDataset(LightningDataset):
+    """A class representing a dataset for VisualPrompt tasks."""
+
+    def __init__(
+        self,
+        task: TaskType | str | None = TaskType.VISUAL_PROMPTING,
```

Comment:
Maybe I missed this part, task name is `VisualPrompt` or visual_prompt but type name in `TaskType` is `VISUAL_PROMPTING`. How about unifying them into either visual prompt or visual prompt"ing"?

Reply:
In v1, the latter (viusal prompting) seems to be set.

Reply:
Ah yes, a bit of a refactoring, and a word with visual_prompting.